Source/Basic-Conditions-And-Behaviors/Runtime/Conditions/CollectedCondition.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using VRBuilder.Core.Attributes;
+using VRBuilder.Core.SceneObjects;
+using UnityEngine;
+using VRBuilder.Core.Properties;
+
+namespace VRBuilder.Core.Conditions
+{
+    /// <summary>
+    /// A condition that checks if new selected objects are entering the selected collider.
+    /// </summary>
+    [DataContract(IsReference = true)]
+    public class CollectedCondition: Condition<CollectedCondition.EntityData>
+    {
+        [DataContract(IsReference = true)]
+        [DisplayName("Collect Condition")]
+        public class EntityData: IConditionData
+        {
+            [HideInProcessInspector]
+            public readonly List<GameObject> Collected = new();
+
+            /// <summary>
+            /// The objects that has to enter the collider.
+            /// </summary>
+            [DataMember]
+            [DisplayName("Object")]
+            public MultipleSceneObjectReference TargetObjects { get; set; }
+
+            /// <summary>
+            /// The collider with trigger to enter.
+            /// </summary>
+            [DataMember]
+            [DisplayName("Collider")]
+            public SingleScenePropertyReference<ColliderWithTriggerProperty> TriggerObject { get; set; }
+
+            public Metadata Metadata { get; set; }
+
+            /// <inheritdoc />
+            [HideInProcessInspector]
+            [IgnoreDataMember]
+            public string Name => $"Collect unique {TargetObjects} in {TriggerObject}";
+            public bool IsCompleted { get; set; }
+
+            public EntityData()
+            {
+                TargetObjects = new MultipleSceneObjectReference();
+                TriggerObject = new SingleScenePropertyReference<ColliderWithTriggerProperty>();
+            }
+        }
+
+        public class CollectedConditionAutocompleter: Autocompleter<EntityData>
+        {
+            public CollectedConditionAutocompleter(EntityData data): base(data)
+            {
+            }
+
+            public override void Complete()
+            {
+            }
+        }
+
+        public class ActivatingProcess: StageProcess<EntityData>
```

Comment:
```suggestion
        public class ActiveProcess: StageProcess<EntityData>
```
Conditions have their logic in the active process, which means they start checking for completion when everything (conditions, behaviors) have finished activating. Your condition works this correctly, as this is returned by the GetActiveProcess method. But it's good to give it the proper name :)